docs/networking.md

```diff
@@ -0,0 +1,114 @@
+<!--[metadata]>
++++
+title = "Networking in Compose"
+description = "How Compose sets up networking between containers"
+keywords = ["documentation, docs,  docker, compose, orchestration, containers, networking"]
+[menu.main]
+parent="smn_workw_compose"
+weight=6
++++
+<![end-metadata]-->
+
+
+# Networking in Compose
+
+By default, Compose sets up a single default [network](http://TODO/docker-networking-docs) for your app. Each container for a service joins the default network and is discoverable via DNS under the service's name.
+
+> **Note:** Your app's network is given the same name as the "project name", which is based on the name of the directory it lives in. See the [CLI docs](cli.md#p-project-name-name) for how to override it.
+
+For example, suppose your app is in a directory called `myapp`, and your `docker-compose.yml` looks like this:
+
+    web:
+      build: .
+      ports:
+        - "8000:8000"
+    db:
+      image: postgres
+
+When you run `docker-compose up`, the following happens:
+
+1. A network called `myapp` is created.
+2. A container is created using `web`'s configuration. It joins the network `myapp` under the name `web`.
+3. A container is created using `db`'s configuration. It joins the network `myapp` under the name `db`.
+
+Each container can now look up the hostname `web` or `db` and get back the appropriate container's IP address. For example, `web`'s application code could connect to the URL `postgres://db:5432` and start using the Postgres database.
+
+Because `web` explicitly maps a port, it's also accessible from the outside world via port 8000 on your Docker host's network interface.
```

Comment:
In 1.8 timeframe is it likely that `ports` will be supported by libnetwork or the plugin multi-host network drivers will support this?  So is this really just an option when using the bridge network?  If so then in the short term I think it would be worth tweaking the wording to finesse this.  e.g. "When testing on a single host system using the bridge network you can..." or similar.

Longer term I think `ports` could map sensibly to plug-in drivers where NAT is required at the boundary of the network.  The drivers could implement this in a variety of ways depending on their architecture. e.g. either via specific NAT device, or via fully distributed NAT similar to what K8s does today, or any other way that fits with the driver.  (This would need some careful consideration though.  Is it a network driver issue?  Or when we have service discovery drivers is it a service discovery driver issue?  Or will there be separate load balancer drivers?  I suggest this is all for discussion later way beyond the scope of this first version of this document.)